sema/harvest/service.py

```
@@ -97,3 +110,6 @@
             log.exception(e)
             log.error("Error running dereference tasks")
             self.error_occurred = True
+        finally:
+            self._result._success = not self.error_occurred
+            return not self.error_occurred
```

Ruff

`return` inside `finally` blocks cause exceptions to be silenced

(B012)